### ISSUE TITLE
Unable to use agentkeepalive with Connection

### ISSUE THREAD

**Issue by jhuntbach-bc** [solana-labs/solana-web3.js#1183]
The `ConnectionConfig` object allows us to provide a `httpAgent` for the `Connection`. If the `endpoint` url starts with `https`, then the agent must be an instance of `https.Agent`. However, this check fails when using the `agentkeepalive` `Agent.HttpsAgent`.
```
Error: The endpoint `https://[...]` can only be paired with an `https.Agent`. You have, instead, supplied an `http.Agent` through `httpAgent`.
```
Is there another way we could enforce this check, which allows us to use `agentkeepalive`?

**Comment by steveluscher** [comment]
Yep! Send me a link to your Stack Exchange question, and I'll answer it there so that others can benefit from the solution going forward!
